packages/chat-headless/tests/chatheadless.clients.test.ts

```diff
@@ -106,6 +108,7 @@ it("handoff http client between event client", async () => {
   await headless.getNextMessage("user message 2");
   expect(botClient.getNextMessage).toHaveBeenCalledTimes(2);
   expect(agentClient.processMessage).toHaveBeenCalledTimes(1);
+  expect(headless.state.conversation.messages.at(-1)?.source).toEqual(MessageSource.BOT);
```

Comment:
I would prefer we have a separate unit test for verify the source instead of adding onto existing tests.